CONTRIBUTING.md

```diff
@@ -12,27 +12,30 @@ elsewhere.
 
 * Code and comments should be formatted to a width no greater than 120 columns
 * Files should be exempt of trailing spaces
-* Each abstraction should live in its own Scala file, i.e `RequestReader.scala`
-* Each implementation should live in the package object, i.e, `io.finch.request`
+* Each abstraction with corresponding implementations should live in its own Scala file, i.e `RequestReader.scala`
+* Each implicit conversion (if possible) should be defined in the corresponding companion object
 
 That said, the Scala style checker `sbt scalastyle` should pass on the code. 
 
 ## Write Tests
-Finch uses [ScalaTest][2] with the following settings:
+Finch uses both [ScalaTest][2] and [ScalaCheck][3] with the following settings:
 
-* Every test should be a `FlatSpec` with `Matchers` mixed in
+* Every test should be a `FlatSpec` with `Matchers` and `Checkers` mixed in
 * An assertion should be done with `x shouldBe y`
+* A property should be checked with `check`
 * Exceptions should be intercepted with `an [Exception] shouldBe thrownBy(x)`
 
 ## Write Docs
-Write clean and simple docs in the `docs.md` file.
+Write clean and simple docs in the `docs` folder.
 
 ## Submit a PR
 * PR should be submitted from a separate branch (use `git checkout -b "fix-123"`)
-* PR should contain only one commit (use `git commit --amend` and `git --force push`)
+* PR should contain only one commit (use `git commit --amend` and `git --force push` or [squash][4] existing commits into one)
```

Comment:
Maybe "generally"? I think there are cases where keeping multiple commits is useful. I could go either way, though.